SparkPost/2web2ui
# FE-931 - Re-style date picker components

[FE-931](https://jira.int.messagesystems.com/browse/FE-931)

### What Changed

- Adds new `getVariantProps` function to the `<Button/>` component that maps the `"variant"` prop to a series of Matchbox `Button` props
- Adds new supporting documentation for the 2web2ui `<Button/>` component
- Update date picker components to use new, Hibana design tokens
- Incorporates default prop for the `<DatePicker/>` component for `id` using `prop-types` instead of assigning a value when destructuring
- Adds a new wrapping `<DatePicker/>` component that passes in `style` props to existing class components while using the `useHibanaOverride` hook
- Updates `<Navbar/>` component in the `<DatePicker/>` component to leverage HTML `<button>` elements instead of clickable icons (as well as `<ScreenReaderOnly/>` content) - makes these controls meet a11y standards
- Updates `<ManualEntryForm/>` component from a style perspective, adding a `style` prop that accepts passed in styles derived from the `useHibanaOverride` hook
- Updates relevant unit tests to match updated components - typically by mocking or passing in `styles` as a prop


### How To Test
- Verify `<Button/>` components in the OG version of the app were not negatively impacted by the introduction of the `variant` prop
- Verify components in the Jira ticket have been updated and leverage the updated Matchbox design tokens
- Verify date pickers in the app continue to behave in the same manor as their counterparts currently in production
- Routes to check:
  * `/reports/summary`
  * `/reports/bounce`
  * `/reports/rejections`
  * `/reports/accepted`
  * `/reports/delayed`
  * `/signals/health-score`
  * `/signals/spam-traps`
  * `/signals/engagement`
  * `/reports/engagement`
  * `/reports/message-events`
  * `/lists/suppressions`
  * `/inbox-placement`
  * `/blacklist/incidents`

### To Do
